streaming/src/message/message_bundle.cc

```diff
@@ -56,34 +56,31 @@ StreamingMessageBundleMetaPtr StreamingMessageBundleMeta::FromBytes(const uint8_
   STREAMING_CHECK(bytes);
 
   uint32_t byte_offset = 0;
-  const uint32_t magic_num = *reinterpret_cast<const uint32_t *>(bytes + byte_offset);
-
-  if (magic_num != StreamingMessageBundleMagicNum) {
-    STREAMING_LOG(INFO) << "Magic Number => " << magic_num;
-  }
-
-  STREAMING_CHECK(magic_num == StreamingMessageBundleMagicNum);
+  STREAMING_CHECK(CheckBundleMagicNum(bytes));
   byte_offset += sizeof(uint32_t);
 
-  uint64_t message_bundle_ts = *reinterpret_cast<const uint64_t *>(bytes + byte_offset);
+  const uint64_t *message_bundle_ts =
+      reinterpret_cast<const uint64_t *>(bytes + byte_offset);
   byte_offset += sizeof(uint64_t);
 
-  uint64_t last_message_id = *reinterpret_cast<const uint64_t *>(bytes + byte_offset);
+  const uint64_t *last_message_id =
+      reinterpret_cast<const uint64_t *>(bytes + byte_offset);
   byte_offset += sizeof(uint64_t);
 
-  uint32_t messageListSize = *reinterpret_cast<const uint32_t *>(bytes + byte_offset);
+  const uint32_t *message_list_size =
+      reinterpret_cast<const uint32_t *>(bytes + byte_offset);
   byte_offset += sizeof(uint32_t);
-  STREAMING_LOG(DEBUG) << "ts => " << message_bundle_ts << " last message id => "
-                       << last_message_id << " message size => " << messageListSize;
+  STREAMING_LOG(DEBUG) << "ts => " << *message_bundle_ts << " last message id => "
+                       << *last_message_id << " message size => " << *message_list_size;
 
-  STREAMING_CHECK(messageListSize <= StreamingConfig::MESSAGE_BUNDLE_MAX_SIZE);
+  STREAMING_CHECK(*message_list_size <= StreamingConfig::MESSAGE_BUNDLE_MAX_SIZE);
 
-  StreamingMessageBundleType messageBundleType =
-      *reinterpret_cast<const StreamingMessageBundleType *>(bytes + byte_offset);
+  const StreamingMessageBundleType *message_bundle_type =
+      reinterpret_cast<const StreamingMessageBundleType *>(bytes + byte_offset);
   byte_offset += sizeof(StreamingMessageBundleType);
 
   auto result = std::make_shared<StreamingMessageBundleMeta>(
-      message_bundle_ts, last_message_id, messageListSize, messageBundleType);
+      *message_bundle_ts, *last_message_id, *message_list_size, *message_bundle_type);
```

Comment:
better to use reference in the constructor of StreamingMessageBundleMeta?
StreamingMessageBundleMeta(uint64_t&, uint64_t&, uint32_t&, StreamingMessageBundleType&);

Reply:
Maybe we should enforce all reference to be const?

Reply:
> better to use reference in the constructor of StreamingMessageBundleMeta?
> StreamingMessageBundleMeta(uint64_t&, uint64_t&, uint32_t&, StreamingMessageBundleType&);



> Maybe we should enforce all references to be const for constructor args?

There is very strange usage :
```cpp
uint32_t a = 1 ,b = 2, c = 3;
StreamingMessageBundleMeta(a, b, c);
```

Or we give two different constructor ：no-reference, all-reference.

Reply:
For simple value type, why not just copy? If you want to change input arguments, using pointer is more explicit than references.

Reply:
We can reduce copy buildin type and use it to deserlize because of it's inplacement in buffer memory already.
Sure, copy a simple type is supporting and right-reference parameters in new constructor is just for compatibility.